searx/search/__init__.py

```diff
@@ -37,18 +40,57 @@ def initialize(settings_engines=None, enable_checker=False, check_network=False,
         initialize_checker()
 
 
+def replace_auto_language(search_query: SearchQuery):
+    """
+    Do nothing except if `search_query.lang` is "auto".
+    In this case:
+    * the value "auto" is replaced by the detected language of the query.
+      The default value is "all" when no language is detected.
+    * `search_query.locale` is updated accordingly
+
+    Use :py:obj:`searx.utils.detect_language` with `only_search_languages=True` to keep
+    only languages supported by the engines.
+    """
+    if search_query.lang != 'auto':
+        return
+
+    detected_lang = detect_language(search_query.query, threshold=0.0, only_search_languages=True)
```

Comment:
@dalf why do we set the threshold to 0.0? .. the default is (was) 0.3 (with 0.0 we get 'en' for a single letter).